schema/types.go

```diff
@@ -88,6 +88,20 @@ type TypedType[T any] interface {
 	SerializeType(data T) (any, error)
 }
 
+// ScalarType is a struct that provides default implementations for ApplyScope and
+// ValidateReferences for types that cannot contain references. Those two
+// functions do not do anything when there are no references.
```

Comment:
The functions don't do anything, on the hardcoded *assumption* that they can't encounter references, so I think the final sentence is a bit misleading! That is, "when there are no references" is a precondition, which is already covered by the first sentence.

Reply:
I removed the second sentence.